pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1660,34 +1661,7 @@ func (volTopology *wcpControllerVolumeTopology) GetTopologyInfoFromNodes(ctx con
 			var selectedSegments []map[string]string
 			for _, topology := range params.TopologyRequirement.GetPreferred() {
 				for label, value := range topology.GetSegments() {
-					var datastores []*cnsvsphere.DatastoreInfo
-					var err error
-					if isPodVMOnStretchedSupervisorEnabled {
-						clusterMorefs, exists := azClustersMap[value]
-						if !exists || len(clusterMorefs) == 0 {
-							return nil, logger.LogNewErrorf(log, "could not find the cluster MoIDs for zone %q in "+
-								"AvailabilityZone resources", value)
-						}
-						for _, clusterMoref := range clusterMorefs {
-							datastoresforClusterMoRef, err := params.Vc.GetDatastoresByCluster(ctx, clusterMoref)
-							if err != nil {
-								return nil, logger.LogNewErrorf(log,
-									"failed to fetch datastores associated with cluster %q", clusterMoref)
-							}
-							datastores = append(datastores, datastoresforClusterMoRef...)
-						}
-					} else {
-						clusterMoref, exists := azClusterMap[value]
-						if !exists || clusterMoref == "" {
-							return nil, logger.LogNewErrorf(log, "could not find the cluster MoID for zone %q in "+
-								"AvailabilityZone resources", value)
-						}
-						datastores, err = params.Vc.GetDatastoresByCluster(ctx, clusterMoref)
-						if err != nil {
-							return nil, logger.LogNewErrorf(log,
-								"failed to fetch datastores associated with cluster %q", clusterMoref)
-						}
-					}
+					datastores := params.TopoSegToDatastoresMap[value]
```

Comment:
Does this give shared datastores if zone is stretched across multiple clusters?

Reply:
The variable `TopoSegToDatastoresMap` is populated when you calculate shared datastores for a zone to pass to CNS for volume provisioning. Therefore, whatever values are calculated & sent to CNS will be stored and used later on to figure out the topology of the selected datastore.